example/project-rails-tailwind/app/components/input_component.html.erb

```diff
@@ -0,0 +1,107 @@
+<% borderColor =
+  @error_message.present? ? "border-error-dark" : "border-background-black-line"
+fontColor =
+  @error_message.present? ? "text-error-dark" : "text-type-black-primary"
+
+isInput = @variant === "input"
+isTextArea = @variant === "textarea"
+iconPresent = @icon && isInput
+type = isInput ? @value : ""
+value = isInput || isTextArea ? @value : "" %>
+
+<div class="flex flex-col align-start gap-[8px]">
+  <div class="flex flex-row items-center justify-between">
+    <%# Label %>
+    <label
+      for="<%=name%>"
+      class="
+        flex
+        flex-row
+        items-center
+        p-0
+        text-base
+        font-normal
+        <%= @disabled ? 'text-type-black-deactivated' : fontColor %>
+      "
+    >
+      <%= label %>
+      <span aria-label="required"><% required ? "*" : "" %></span>
+    </label>
+
+    <%# Hint Text %>
+    <% if @hintText %>
+      <p
+        class="paragraph-2 <%= @disabled ? 'text-type-black-deactivated' : 'text-type-black-secondary' %>"
+      >
+        <%= @hintText %>
+      </p>
+    <% end %>
+  </div>
+
+  <div
+    class="
+      flex
+      flex-row
+      relative
+      border
+      focus-within:border-button-active
+      focus-within:w-full
+      <%= borderColor %>
+      <%= @error_message.present? ? 'border-error-dark' : '' %>
+      <%= @disabled ? 'bg-button-deactivatedBackground border-type-black-deactivated' : 'text-type-black-primary' %>
+    "
+  >
+    <%# Input / Text Area %>
+    <input
+      class="
+        peer
+        focus:outline-none
+        text-sm
+        py-[14px]
+        px-[16px]
+        w-full
+        <%= iconPresent && 'pl-[48px]' %>
+        <%= isTextArea ? 'resize-none' : '' %>
+      "
+      disable="<%= @disabled %>"
+      placeholder="<%= @placeholder %>"
+      name="<%= @name %>"
+      required="<%= @required %>"
+      type="<%= @type %>"
+    />
+
+    <%# Icon %>
+    <% if iconPresent %>
+      <%= tag.i class:
+              "absolute pointer-events-none left-[16px] top-[14px] peer-disabled:text-type-black-deactivated text-type-black-primary",
+            data: {
+              feather: "search"
```

Comment:
The story for input has a selector for different icons in place of the search - doesn't appear to be implemented here. I'd either remove those options from the story, or implement that functionality. As these are meant to be examples of how to use Shipwright Tokens with this stack, I'd just go with whatever was easiest.

Reply:
I removed all the bloated variants.